pep-0483.txt

```diff
@@ -38,276 +70,285 @@ Summary of gradual typing
-   disappears from the argument list, e.g. Intersection[int, str, Any]
-   == Intersection[int, str]. Intersection[Any, object] is object. The
-   interaction between Intersection and Union is complex but should be
-   no surprise if you understand the interaction between intersections
-   and unions in set theory (note that sets of types can be infinite in
-   size, since there is no limit on the number of new subclasses).
+   ``Intersection[int, Intersection[float, str]] == Intersection[int,
+   float, str]``. An intersection of fewer types is a supertype of an
+   intersection of more types, e.g. ``Intersection[int, str]`` is a supertype
+   of ``Intersection[int, float, str]``. An intersection of one argument is
+   just that argument, e.g. ``Intersection[int]`` is ``int``.
+   When argument have a subtype relationship, the more specific type
+   survives, e.g. ``Intersection[str, Employee, Manager]`` is
+   ``Intersection[str, Manager]``. ``Intersection[]`` is illegal,
+   so is ``Intersection[()]``. Corollary: ``Any``
+   disappears from the argument list, e.g. ``Intersection[int, str, Any]
+   == Intersection[int, str]``. ``Intersection[Any, object]`` is ``object``.
+   The interaction between ``Intersection`` and ``Union`` is complex but
+   should be no surprise if you understand the interaction between
+   intersections and unions in set theory (note that sets of types can be
+   infinite in size, since there is no limit on the number of new subclasses).
 
 
-Pragmatics
-----------
+Generic types
+=============
 
-Some things are irrelevant to the theory but make practical use more
-convenient. (This is not a full list; I probably missed a few and some
-are still controversial or not fully specified.)
+TODO: Explain motivation, generics as functors (but avoid this word),
+examples, necessity of type variables and declaration syntax that
+described below.
 
-- Type aliases, e.g.
 
-  *  Point = Tuple[float, float]
-  *  def distance(p: Point) -> float: ...
+Type variables
+--------------
 
-- Forward references via strings, e.g.
+(TODO: Explain more. See also the `mypy docs on
+generics <http://mypy.readthedocs.io/en/latest/generics.html>`_.)
 
-  * class C:
-  
-    + def compare(self, other: 'C') -> int: ...
+- ``X = TypeVar('X')``. Declares a unique type variable. The name must match
+  the variable name.
 
-- If a default of None is specified, the type is implicitly Optional, e.g.
+- ``Y = TypeVar('Y', t1, t2, ...)``. Ditto, constrained to ``t1``, etc. Behaves
+  like ``Union[t1, t2, ...]`` for most purposes, but when used as a type
+  variable, subclasses of ``t1``, etc. are replaced by the most-derived base
+  class among ``t1``, etc. (TODO: reformulate this)
 
-  *  def get(key: KT, default: VT = None) -> VT: ...
+- Example of constrained type variables::
 
-- Don't use dynamic type expressions; use builtins and imported types
-  only. No 'if'.
+    AnyStr = TypeVar('AnyStr', str, bytes)
```

Comment:
I wonder if the example would gain clarity by renaming this variable to something else (maybe ``S``), conforming to the naming conventions for type variables.